x-pack/legacy/plugins/alerting/server/plugin.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Hapi from 'hapi';
+import { Logger, Services } from './types';
+import { SpacesPlugin } from '../../spaces';
+import { ActionsPlugin } from '../../actions';
+import { AlertsClient } from './alerts_client';
+import { TaskManager } from '../../task_manager';
+import { AlertTypeRegistry } from './alert_type_registry';
+import { XPackMainPlugin } from '../../xpack_main/xpack_main';
+import { AlertsClientFactory } from './lib';
+import { KibanaRequest, SavedObjectsLegacyService } from '../../../../../src/core/server';
+import { EncryptedSavedObjectsPlugin } from '../../encrypted_saved_objects';
+import { ElasticsearchPlugin } from '../../../../../src/legacy/core_plugins/elasticsearch';
+import { PluginSetupContract as SecurityPluginSetupContract } from '../../../../plugins/security/server';
+import {
+  createAlertRoute,
+  deleteAlertRoute,
+  findAlertRoute,
+  getAlertRoute,
+  listAlertTypesRoute,
+  updateAlertRoute,
+  enableAlertRoute,
+  disableAlertRoute,
+  updateApiKeyRoute,
+  muteAllAlertRoute,
+  unmuteAllAlertRoute,
+  muteAlertInstanceRoute,
+  unmuteAlertInstanceRoute,
+} from './routes';
+
+interface PluginInitializerContext {
+  logger: {
+    get: () => Logger;
+  };
+}
+
+interface CoreSetup {
```

Comment:
I'd suggest you call this `AlertingCoreSetup` so that it's clear from your code that this isn't the NP-compatible `CoreSetup` but a custom shim. (same thing for `PluginInitializerContext` and `PluginsSetup`

Reply:
Fixed in https://github.com/elastic/kibana/pull/46954/commits/68a6bcd25ec3a67e555d9111e554011abed678fe.